### PR TITLE
update racket project type with :install and :package

### PR DESCRIPTION
**Pull request by xgqt**
This commit adds:
- install - which runs "raco pkg install" to insatll the current package
- package - which runs raco pkg create --source X
  where X is the absolute path to the current package directory

Signed-off-by: Maciej Barć <xgqt@riseup.net>

-----------------

- [ ] The commits are consistent with our [contribution guidelines](../blob/master/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing ([`eldev test`](https://github.com/doublep/eldev))
- [ ] The new code is not generating bytecode or `M-x checkdoc` warnings
- [ ] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -3054,7 +3054,9 @@ test/impl/other files as below:
 ;; Racket
 (projectile-register-project-type 'racket '("info.rkt")
                                   :project-file "info.rkt"
-                                  :test "raco test .")
+                                  :test "raco test ."
+                                  :install "raco pkg install"
+                                  :package (concat "raco pkg create --source " (expand-file-name ".")))
```
Comment (bbatsov) [inline projectile.el:3059]: I don't think that `expand-file-name` is a good idea, as it's just going to be evaluated when Projectile loads. I guess a simple `.` is what you need.

**Comment by xgqt** [comment]
`:package` doesn't work as I intended, it is executed as `"raco pkg create --source <user's home folder>"`

I tried changing to
`:package (concat "raco pkg create --source " (projectile-project-root))`
but this shows only `"raco pkg create --source"` when trying to execute `projectile-package-project`

File: projectile.el (hunk 1 above)
Comment (xgqt) [reply projectile.el:3059]: > I don't think that `expand-file-name` is a good idea, as it's just going to be evaluated when Projectile loads. I guess a simple `.` is what you need.

yea, that the catch, it has to be absolute path

Comment (xgqt) [reply projectile.el:3059]: image


image

**Comment by xgqt** [comment]
changed `:package` to `raco pkg create --source $(pwd)`

**Comment by bbatsov** [comment]
You can't have dynamic code evaluation here, as this is simply evaluated once and becomes a string. It's not something that's going to be evaluated again and again. But I guess you already figured this out. 

I don't get how is `$(pwd)` different from `.`.

**Comment by xgqt** [comment]
> You can't have dynamic code evaluation here, as this is simply evaluated once and becomes a string. It's not something that's going to be evaluated again and again. But I guess you already figured this out.
> 
> I don't get how is `$(pwd)` different from `.`.

It is different in this context, as raco will error on `.`

https://github.com/bbatsov/projectile/pull/1667#discussion_r638488722

**Comment by xgqt** [comment]
Thanks for cooperation!

**Pull request merged**